mcp/mcp.php

```diff
@@ -0,0 +1,560 @@
+include_once XOOPS_ROOT_PATH . '/mcp/prompts.php';
+
+class FormulizeMCP
+{
+	use tools;
+	use resources;
+	use prompts;
+
+	private $config;
+	private $db;
+	public $enabled = false;
+	public $canBeEnabled = false;
+	public $tools;
+	public $resources;
+	public $prompts;
+	private $authenticatedUser = null;
+	private $authenticatedUid = 0;
+	public $userGroups = array();
+	public $baseUrl;
+	private $mcpRequest = array();
+
+	public function __construct($config = null)
+	{
+		// Front load
+		$this->db = $this->getFormulizeDatabase();
+		$authHeader = $this->getAuthorizationHeader();
+
+		if ($authHeader == 'Bearer test-header-passthrough-check') {
+			$this->canBeEnabled = true;
+		}
+
+		if (isMCPServerEnabled()) {
+			$this->enabled = true;
+			$this->canBeEnabled = true;
+		}
+
+		// Get the raw http request contents
+		$input = file_get_contents('php://input');
+		// @todo should we throw if the input is empty?
+		$request = !empty($input) ? json_decode($input, true) : '';
+		$this->mcpRequest = [
+			'id' => $request['id'] ?? null,
+			'method' => $request['method'] ?? '',
+			'params' => $request['params'] ?? [],
+			'localServerName' => strtolower(FormulizeObject::sanitize_handle_name($request['localServerDetails']['name'] ?? 'formulize'))
+		];
+		$this->config = $config ?: $this->getDefaultConfig();
+		$this->baseUrl = $this->getBaseUrl();
+
+		// Register tools, resources, and prompts
+		$this->registerTools();
+		$this->registerResources();
+		$this->registerPrompts();
+	}
+
+	/**
+	 * Send an HTTP response with headers and body
+	 *
+	 * @param array $body The response body, either JSON or HTML
+	 * @param int $httpResponseCode The HTTP response code to send (default 200)
+	 * @param array $headers Additional headers to set (default empty array)
+	 * @param bool $isJSON Whether the body is JSON (default true)
+	 * @return void
+	 */
+	public function sendResponse(array|string $body, int $httpResponseCode = 200, array $headers = [], bool $isJSON = true) {
+		// Iterate throug the headers and set them
+		foreach ($headers as $header) {
+			header($header);
+		}
+		// Prevent ALL caching at every level
+		header('Cache-Control: no-cache, no-store, must-revalidate, max-age=0');
+		header('Pragma: no-cache');
+		header('Expires: Thu, 01 Jan 1970 00:00:00 GMT');
+		header('Last-Modified: ' . gmdate('D, d M Y H:i:s') . ' GMT');
+		header('ETag: "' . uniqid() . '"');
+		header('Vary: *');
+
+		// CORS headers
+		header('Access-Control-Allow-Origin: *');
+		header('Access-Control-Allow-Methods: GET, POST, OPTIONS');
+		header('Access-Control-Allow-Headers: Content-Type, Authorization');
+		header('Access-Control-Max-Age: 0');
+
+		// Set the HTTP response code
+		http_response_code($httpResponseCode);
+
+		// Set the content type to JSON
+		if ($isJSON) {
+			header('Content-Type: application/json; charset=utf-8');
+			// Encode the body as JSON
+			echo json_encode($body, JSON_PRETTY_PRINT);
```

Comment:
In general, it seems that everything lower down is doing some variation on this...

```
eturn [
			'jsonrpc' => '2.0',
			'result' => [
				'prompts' => array_values($this->prompts)
			],
			'id' => $id
];
```

What I'm concerned about is the jsonrpc and result structure, and id. If everything always returns stuff with this overall basic structure, should we put that here instead, and the lower methods would return only the result part or something like that? There may be reasons I'm not aware of why that's a bad idea. I'm just thinking of it from the point of view of avoiding repetition and right now we're repeating a lot of this stuff over and over in the lower methods.